### PR TITLE
[Test] Performance benchmarks for DGL kernels

### PR DESCRIPTION
**Pull request by jermainewang**
## Description

* gspmm with and without broadcasting (copy_u + sum/max, u_mul_e + sum)
* gsddmm with and without broadcasting (u_dot_v)
* Add utilities for benchmarking GLOPS.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: benchmarks/benchmarks/kernel/bench_gspmm_u_mul_e_sum.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+from .. import utils
+
+def calc_gflops(graph, feat_size, num_heads, time):
+    return round(2 * graph.num_edges() * feat_size / 1000000000 / time, 2)  # count both mul and add
+
+@utils.benchmark('flops', timeout=600)
+@utils.parametrize('graph', ['ogbn-arxiv', 'reddit', 'ogbn-proteins'])
+@utils.parametrize('feat_size', [4, 32, 256])
+@utils.parametrize('num_heads', [0, 1, 4])
```
Comment (yzh119) [inline benchmarks/benchmarks/kernel/bench_gspmm_u_mul_e_sum.py:13]: What do you mean by 0 head?

File: benchmarks/benchmarks/kernel/bench_gsddmm_u_dot_v.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import torch
+
+from .. import utils
+
+def calc_gflops(graph, feat_size, num_heads, time):
+    return round(2 * graph.num_edges() * feat_size / 1000000000 / time, 2)  # count both mul and add
+
+@utils.benchmark('flops', timeout=600)
+@utils.parametrize('graph', ['ogbn-arxiv', 'reddit', 'ogbn-proteins'])
+@utils.parametrize('feat_size', [4, 32, 256])
```
Comment (yzh119) [inline benchmarks/benchmarks/kernel/bench_gsddmm_u_dot_v.py:12]: We should clarify the definition of `feat_size` (especially when number of heads is not 1).

**Review by yzh119 — APPROVED** [review]
LGTM.

File: benchmarks/benchmarks/kernel/bench_gspmm_u_mul_e_sum.py (hunk 1 above)
Comment (jermainewang) [reply benchmarks/benchmarks/kernel/bench_gspmm_u_mul_e_sum.py:13]: added description

File: benchmarks/benchmarks/kernel/bench_gsddmm_u_dot_v.py (hunk 2 above)
Comment (jermainewang) [reply benchmarks/benchmarks/kernel/bench_gsddmm_u_dot_v.py:12]: added description

**Pull request merged**